src/main/java/gregtech/integration/theoneprobe/provider/CoverProvider.java

```diff
@@ -0,0 +1,233 @@
+import gregtech.api.util.GTUtility;
+import gregtech.common.covers.*;
+import gregtech.common.covers.filter.*;
+import mcjty.theoneprobe.api.IProbeHitData;
+import mcjty.theoneprobe.api.IProbeInfo;
+import mcjty.theoneprobe.api.TextStyleClass;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.util.text.TextFormatting;
+import net.minecraftforge.common.capabilities.Capability;
+
+import javax.annotation.Nonnull;
+
+public class CoverProvider extends CapabilityInfoProvider<ICoverable> {
+
+    @Nonnull
+    @Override
+    protected Capability<ICoverable> getCapability() {
+        return GregtechTileCapabilities.CAPABILITY_COVERABLE;
+    }
+
+    @Override
+    public String getID() {
+        return GTValues.MODID + ":coverable_provider";
+    }
+
+    @Override
+    protected void addProbeInfo(@Nonnull ICoverable capability, @Nonnull IProbeInfo probeInfo, @Nonnull EntityPlayer player, @Nonnull TileEntity tileEntity, @Nonnull IProbeHitData data) {
+        CoverBehavior coverBehavior = capability.getCoverAtSide(data.getSideHit());
+        if (coverBehavior instanceof CoverConveyor) {
+            conveyorInfo(probeInfo, (CoverConveyor) coverBehavior);
+        } else if (coverBehavior instanceof CoverPump) {
+            pumpInfo(probeInfo, (CoverPump) coverBehavior);
+        } else if (coverBehavior instanceof CoverItemFilter) {
+            itemFilterInfo(probeInfo, (CoverItemFilter) coverBehavior);
+        } else if (coverBehavior instanceof CoverFluidFilter) {
+            fluidFilterInfo(probeInfo, (CoverFluidFilter) coverBehavior);
+        }
+    }
+
+    /**
+     * Displays text for {@link CoverConveyor} related covers
+     *
+     * @param probeInfo the info to add the text to
+     * @param conveyor  the conveyor to get data from
+     */
+    private static void conveyorInfo(@Nonnull IProbeInfo probeInfo, @Nonnull CoverConveyor conveyor) {
+        String rateUnit = " {*cover.conveyor.transfer_rate*}";
+
+        if (conveyor instanceof CoverItemVoiding) {
+            itemVoidingInfo(probeInfo, (CoverItemVoiding) conveyor);
+        }
+
+        // do not display the regular rate if the cover has a specialized rate
+        if (!(conveyor instanceof CoverItemVoiding) && (!(conveyor instanceof CoverRoboticArm) || ((CoverRoboticArm) conveyor).getTransferMode() == TransferMode.TRANSFER_ANY)) {
+            transferRateText(probeInfo, conveyor.getConveyorMode(), rateUnit, conveyor.getTransferRate());
+        }
+
+        ItemFilterContainer filter = conveyor.getItemFilterContainer();
+        if (conveyor instanceof CoverRoboticArm) {
+            CoverRoboticArm roboticArm = (CoverRoboticArm) conveyor;
+            transferModeText(probeInfo, roboticArm.getTransferMode(), rateUnit, filter.getTransferStackSize(), filter.getFilterWrapper().getItemFilter() != null);
+        }
+        itemFilterText(probeInfo, filter.getFilterWrapper().getItemFilter());
```

Comment:
This displays `Filter:` for an OreDictionaryFilter when the text box is empty, while nothing is displayed for a regular empty item filter. If the text box is empty for the OreDictionaryFilter, this should also be hidden.